http/client/src/main/java/org/wildfly/security/http/client/ElytronHttpClient.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.http.client;
+
+import org.wildfly.security.http.client.exception.ElytronHttpClientException;
+import org.wildfly.security.http.client.mechanism.basic.ElytronHttpClientBasicAuthMechanism;
+import org.wildfly.security.http.client.mechanism.bearer.ElytronHttpClientBearerAuthMechanism;
+import org.wildfly.security.http.client.mechanism.digest.ElytronHttpClientDigestAuthMechanism;
+import org.wildfly.security.http.client.utils.ElytronHttpClientCredentialUtils;
+import org.wildfly.security.http.client.utils.ElytronMessages;
+
+import javax.net.ssl.SSLContext;
+
+import static org.wildfly.security.http.HttpConstants.OK;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Elytron client for HTTP authentication
+ *
+ * @author <a href="mailto:kekumar@redhat.com">Keshav Kumar</a>
+ */
+public class ElytronHttpClient {
+
+    private HttpClient httpClient;
+    private static ElytronHttpClientCredentialUtils elytronHttpClientCredentialProvider = new ElytronHttpClientCredentialUtils();
+
+    public ElytronHttpClient() {
+        this.httpClient = HttpClient.newHttpClient();
+    }
+
+    private HttpResponse getResponse(HttpRequest request) throws IOException, InterruptedException{
+        return httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+    }
+
+    private void addSSLContextToHttpCient(URI uri){
+        SSLContext sslContext = elytronHttpClientCredentialProvider.getSSLContext(uri);
+        httpClient = HttpClient.newBuilder().sslContext(sslContext).build();
+    }
+    private HttpRequest evaluateMechanism(URI uri) {
+        HttpRequest request = HttpRequest
+                .newBuilder()
```

Comment:
@keshav-725 The request we are sending is always a GET request ( https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpRequest.Builder.html ) . We have to think about how a user of this API will be able to send a `POST` or `PUT` requests with some body as well, and add a needed methods for that. 

To achieve this, we will probably have to add a new method, for example `send(HttpRequest request)` where we allow a user of this API to configure a POST method with a body and some headers.

Then the ElytronHttpClient will add a needed authentication configuration but the rest of the request (request body etc) will be in the hands of the user